cloudsync_onedrive.py

```diff
@@ -340,6 +340,21 @@ def _set_drive_list(self):
             log.error("failed to get personal drive info: %s", repr(e))
             raise CloudTokenError("Invalid account, or no onedrive access")
 
+        #shared items
+        shared = self._direct_api("get", "/me/drive/sharedWithMe")
+        for item in shared.get("value", []):
+            try:
+                if not "folder" in item:
+                    # ignore shared files
+                    continue
+                drive_id = item["remoteItem"]["parentReference"]["driveId"]
+                path = urllib.parse.urlparse(item["webUrl"]).path.split('/')
```

Comment:
I'm wondering if we need to do url decoding here

Reply:
Line 363 too

Reply:
added url decoding